### DISCUSSION TITLE
Can you smoke test production deploys with Github Actions?

### DISCUSSION THREAD

**Discussion by aegatlin** [vercel/next.js#11290] [Help] [answered]
I want to use Github Actions to try and run a smoke test against prod, triggered by a successful deployment to production. I know that Github Actions has a `on: deployment` and `on: deployment_status` trigger, but it only seems to run at the **beginning** of a deploy. I want a trigger that occurs on **deployment success**. Does anyone know how to do this? Are there any examples of how to do this?

(Incidentally, in investigating, I noticed that Nextjs doesn't utilize the notion of smoke tests in their cicd github action :P)

**Comment by Timer** [comment, marked as answer]
The `on: deployment_status` trigger should be what you're looking for (this should be triggered multiple times, e.g. once for pending, once for success, etc).

https://help.github.com/en/actions/reference/events-that-trigger-workflows#deployment-status-event-deployment_status

> deployment URL is available under the webhook payload as target_url after the state is success

**Comment by mikenikles** [comment]
@aegatlin Did you get this to work? I've just opened another discussion because that `on: deployment_status` trigger doesn't seem to work.

See https://github.com/vercel/vercel/discussions/4658.
